test/utransport/uattributes_test.cpp

```diff
@@ -53,6 +60,45 @@ TEST(UAttributesTest, Class)
     EXPECT_EQ(nonEmptyAttributes.priority(), priority);
 }
 
+// Test building a UAttributes object used for request and response
+TEST(UAttributesTest, BuildingRequestResponse) 
+{
+    // Return address
+    auto source = BuildUUri()
+                    .setEntity(BuildUEntity().setName("hartley_app").setMajorVersion(1).build())
+                    .setResource(BuildUResource().setRpcResponse().build())
+                    .build();
+    
+    // Method we want to invoke
+    UUri sink = BuildUUri()
+                    .setEntity(BuildUEntity().setName("test_service").setMajorVersion(1).build())
+                    .setResource(BuildUResource().setRpcRequest("test_function").build())
+                    .build();
+
+    // Create a UAttributes object with some values
+    UAttributes request = UAttributesBuilder().request(source, sink, UPriority::UPRIORITY_CS4, 1000).build();
+
+    // Test getters for the attributes
+    EXPECT_EQ(request.sink().entity().name(), "test_service");
+    EXPECT_EQ(request.sink().resource().name(), "rpc");
+    EXPECT_EQ(request.sink().resource().instance(), "test_function");
+    EXPECT_EQ(request.ttl(), 1000);
+    EXPECT_EQ(request.source().entity().name(), "hartley_app");
+    EXPECT_EQ(request.source().resource().name(), "rpc");
+    EXPECT_EQ(request.source().resource().id(), 0);
+    EXPECT_EQ(request.source().resource().instance(), "response");
```

Comment:
Nitpick: Swap these two lines - makes the sink and source testing more similar.

Reply:
done